.travis.yml

```diff
@@ -159,8 +159,8 @@ script:
   - export CMAKE_ARGS="-DLUA_LIBRARY=${LUALIBRARY} -DLUA_INCLUDE_DIR=${LUAINCLUDE} -D OVERRIDE_VERSION=$AWESOME_VERSION -DSTRICT_TESTS=true"
   - |
     if [ "$DO_COVERAGE" = "codecov" ]; then
-      export CFLAGS="-fprofile-arcs -ftest-coverage"
-      export LDFLAGS="-fprofile-arcs"
+      export CFLAGS="-fprofile-arcs -ftest-coverage -O0"
+      export LDFLAGS="-fprofile-arcs -O0"
```

Comment:
Are you sure this works? I just asked Google and it seems like this is only added to CMake's default flags (AFAIR this includes `-O2`). However, I could not figure out if these values are appended or prepended to CMake's default flags. If `-O0` ends up being before the default `-O2`, it won't have any effect.

Also, the commit message could say a bit more about why this is done. I guess the reason is to have more meaningful coverage data (so that things that the optimizer removes or moves around don't end up confusing us), but I feel like that could still be in there.